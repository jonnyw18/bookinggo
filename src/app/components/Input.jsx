import React from 'react';


const Input = ({
   handleChange,
   searchData,
   handleOnFocus,
   handleOnBlur,
   showResults}) => {
    return (
        <React.Fragment>
            <input
                type="text"
                onFocus={handleOnFocus}
                onBlur={handleOnBlur}
                onChange={handleChange}
            />
            <div hidden={showResults}>
                {
                    searchData.length > 0
                        ? searchData.map((locations, index) => (
                            locations.city
                                ? <p key={index}> {locations.city} </p>
                                : <p key={index}> {locations.name} </p>
                        ))
                        : null
                }
            </div>


        </React.Fragment>
    )
};

export default Input