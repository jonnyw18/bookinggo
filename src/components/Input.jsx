import React from 'react';
import Loading from "./Loading";

const Input = ({
   handleChange,
   searchData,
   handleOnFocus,
   handleOnBlur,
   showResults,
   userInput,
   loading
}) => {
    return (
        <div className="search-container">
            <h1> Let's find your ideal car </h1>
            <label> Pick-up Location </label>
            <div className="input-container">
                <input
                    type="text"
                    onFocus={handleOnFocus}
                    onBlur={handleOnBlur}
                    onChange={handleChange}
                    placeholder="city, airport, station, region, districtâ€¦"
                />
                <div className="loading-spinner-container">
                    {
                        userInput.length > 1
                            ? loading
                            ? <Loading />
                            : null
                            : null
                    }
                </div>

            </div>
            <div hidden={showResults} className="test-scss">
                {
                    searchData.length > 0
                        ? searchData.map((locations, index) => (
                            locations.city
                                ? <p key={index}> {locations.city} </p>
                                : <p key={index}> {locations.name} </p>
                        ))
                        : null
                }
            </div>
        </div>
    )
};

export default Input