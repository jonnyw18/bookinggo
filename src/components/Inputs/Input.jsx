import React from 'react';
import PropTypes from 'prop-types';
import { Loading } from "../";

const Input = ({
   handleChange,
   handleOnFocus,
   handleOnBlur,
   userInput,
   loading
}) =>
    <div className="search-container">
        <h2> Let's find your ideal car </h2>
        <label> Pick-up Location </label>
        <div className="input-container">
            <input
                id="locations-input"
                type="text"
                onFocus={handleOnFocus}
                onBlur={handleOnBlur}
                onChange={handleChange}
                placeholder={
                    window.innerWidth < 600
                        ? "e.g. town, city or airport"
                        : "city, airport, station, region, districtâ€¦"
                }
            />
            <div className="loading-spinner-container">
                {
                    userInput.length > 1
                        ? loading
                            ? <Loading />
                            : null
                        : null
                }
            </div>
        </div>
    </div>;

Input.propTypes = {
    handleChange: PropTypes.func.isRequired,
    handleOnFocus: PropTypes.func.isRequired,
    handleOnBlur: PropTypes.func.isRequired,
    userInput: PropTypes.string.isRequired,
    loading: PropTypes.bool.isRequired
};

export default Input