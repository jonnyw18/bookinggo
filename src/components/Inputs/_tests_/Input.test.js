import Input from '../Input';

describe('<Input />', () => {
    it('renders correctly', () => {
      const wrapper = shallow(
        <Input
            handleChange={() => {}}
            handleOnFocus={() => {}}
            handleOnBlur={() => {}}
            userInput={''}
            loading={true}
        />
    );
    expect(wrapper).toMatchSnapshot();
  });

    it('Formats header text correctly', () => {
      const wrapper = render(
            <Input
                handleChange={() => {}}
                handleOnFocus={() => {}}
                handleOnBlur={() => {}}
                userInput={''}
                loading={true}
            />
       );
       const text = wrapper.find('h2').text();
       expect(text).toEqual(' Let\'s find your ideal car ');
    });

    it('Calls handleOnFocus when user clicks on input', () => {
        const spy = sinon.spy();
        const wrapper = mount(
            <Input
                handleChange={() => {}}
                handleOnFocus={spy}
                handleOnBlur={() => {}}
                userInput={''}
                loading={true}
            />
        );
        const input = wrapper.find('input');

        input.simulate('focus');
        expect(spy.calledOnce).toBe(true);
    });

    it('Calls handleOnBlur when user clicks away from input', () => {
        const spy = sinon.spy();
        const wrapper = mount(
            <Input
                handleChange={() => {}}
                handleOnFocus={() => {}}
                handleOnBlur={spy}
                userInput={''}
                loading={true}
            />
        );
        const input = wrapper.find('input');

        input.simulate('blur');
        expect(spy.calledOnce).toBe(true);
    });

    it('Calls handleChange when user types into input', () => {
        const spy = sinon.spy();
        const wrapper = mount(
            <Input
                handleChange={spy}
                handleOnFocus={() => {}}
                handleOnBlur={() => {}}
                userInput={''}
                loading={true}
            />
        );
        const input = wrapper.find('input');

        input.simulate('change', { value: 'Change' });
        expect(spy.calledOnce).toBe(true);
    })
});
