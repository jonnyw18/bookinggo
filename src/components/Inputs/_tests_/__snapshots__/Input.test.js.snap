// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<div
  className="search-container"
>
  <h2>
     Let's find your ideal car 
  </h2>
  <label>
     Pick-up Location 
  </label>
  <div
    className="input-container"
  >
    <input
      id="locations-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="city, airport, station, region, districtâ€¦"
      type="text"
    />
    <div
      className="loading-spinner-container"
    />
  </div>
</div>
`;
